Sources/SourceKitLSP/Swift/OpenInterface.swift

```diff
@@ -16,82 +16,83 @@ import LanguageServerProtocol
 import SKSupport
 import SourceKitD
 
-struct InterfaceInfo {
+struct GeneratedInterfaceInfo {
   var contents: String
 }
 
 extension SwiftLanguageService {
-  public func openInterface(_ request: OpenInterfaceRequest) async throws -> InterfaceDetails? {
-    let uri = request.textDocument.uri
-    let moduleName = request.moduleName
+  public func openGeneratedInterface(
+    _ request: OpenGeneratedInterfaceRequest
+  ) async throws -> GeneratedInterfaceDetails? {
     let name = request.name
     let symbol = request.symbolUSR
     let interfaceFilePath = self.generatedInterfacesPath.appendingPathComponent("\(name).swiftinterface")
     let interfaceDocURI = DocumentURI(interfaceFilePath)
     // has interface already been generated
     if let snapshot = try? self.documentManager.latestSnapshot(interfaceDocURI) {
```

Comment:
Just to make sure I understand, once the client has opened the interface, it gets populated in the `documentManager`?

Reply:
Yes, the client should send a `textDocument/didOpen` notification to SourceKit-LSP, which will then add it to the document manager like any other source file.